map/local_ads_manager.cpp

```diff
@@ -48,6 +61,50 @@ std::chrono::steady_clock::time_point Now()
 {
   return std::chrono::steady_clock::now();
 }
+
+#ifdef TEMPORARY_LOCAL_ADS_JSON_SERIALIZATION
+struct FreeDeletor
```

Comment:
We already have FreeDeletor in search/search_quality/sample.cpp. Please, move this class to the base and reuse in both places.